portal/server/controllers/utilisation-report-service/utilisation-report-upload/utilisation-report-status.js

```diff
@@ -0,0 +1,39 @@
+const { addBusinessDaysWithHolidays } = require('../../../helpers/addBusinessDays');
+const api = require('../../../api');
+
+/**
+ * Calls the Gov UK bank holidays API to get a list of public holidays, uses only
+ * events for England and Wales and maps to date.
+ * @returns {Promise} - List of dates as an array.
+ */
+const getBankHolidays = async (userToken) => {
+  const bankHolidaysFromGovApi = await api.getUkBankHolidays(userToken);
+  return bankHolidaysFromGovApi['england-and-wales'].events.map((event) => new Date(event.date));
+};
+
+/**
+ * Gets the reporting period based off the 1st day of the current month.
+ * @param {Date} firstDayOfCurrentMonth - The value to test for being a number.
+ * @returns {string} - Previous Month (long) and Year (numeric) as a string.
+ */
+const getCurrentReportPeriod = (firstDayOfCurrentMonth) => {
+  const reportDate = new Date(firstDayOfCurrentMonth.getFullYear(), firstDayOfCurrentMonth.getMonth() - 1, 1);
+  return reportDate.toLocaleDateString('en-GB', { year: 'numeric', month: 'long' });
+};
+
+/**
+ * Gets the current report due date based off the 1st day of the current month.
+ * @param {Date} firstDayOfCurrentMonth - The value to test for being a number.
+ * @returns {Promise} - Due Date (numeric), Month (long) and Year (numeric) as a string.
+ */
+const getCurrentReportDueDate = async (firstDayOfCurrentMonth, userToken) => {
+  const bankHolidays = await getBankHolidays(userToken);
+  const dateAdd10BusinessDays = addBusinessDaysWithHolidays(firstDayOfCurrentMonth, 10, bankHolidays);
+  return dateAdd10BusinessDays.toLocaleDateString('en-GB', { year: 'numeric', month: 'long', day: 'numeric' });
+};
```

Comment:
I can see why you're passing in `firstDayOfCurrentMonth`, but I feel like you could change this so that `getCurrentReportPeriod` doesn't take any input arguments as it will then be easier to reuse (same idea for `getCurrentReportDueDate`). You are using the same current date for both functions so I just think it may make the `utilisation-report-upload` controller above a little easier to read if all the date based logic is done within these functions. Happy with them as they are too though! :)